userspace/falco/app/actions/print_support.cpp

```diff
@@ -37,23 +39,61 @@ falco::app::run_result falco::app::actions::print_support(falco::app::state& s)
 	if(s.options.print_support)
 	{
 		nlohmann::json support;
-		struct utsname sysinfo;
 		std::string cmdline;
 
+#ifndef _WIN32
+		struct utsname sysinfo;
 		if(uname(&sysinfo) != 0)
 		{
 			return run_result::fatal(std::string("Could not uname() to find system info: ") + strerror(errno));
 		}
+#else
+		OSVERSIONINFO osvi;
+		SYSTEM_INFO sysInfo;
+		TCHAR computerName[256];
+		DWORD size = sizeof(computerName);
+
+		osvi.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
+		GetSystemInfo(&sysInfo);
+		if(!GetVersionEx(&osvi) || !GetComputerName(computerName, &size))
+		{
+			return run_result::fatal(std::string("Could not get system info: ") + strerror(errno));
+		}
+#endif
 
 		const falco::versions_info infos(s.offline_inspector);
 		support["version"] = infos.falco_version;
 		support["engine_info"] = infos.as_json();
 
+#ifndef _WIN32
 		support["system_info"]["sysname"] = sysinfo.sysname;
 		support["system_info"]["nodename"] = sysinfo.nodename;
 		support["system_info"]["release"] = sysinfo.release;
 		support["system_info"]["version"] = sysinfo.version;
 		support["system_info"]["machine"] = sysinfo.machine;
+#else
+		support["system_info"]["sysname"] = "Windows";
```

Comment:
Wouldn't it be a little cleaner to create a function that is able to output the `system_info`, have one for linux/mac and one for Windows so we can better separate the two implementations?

Reply:
That makes sense! 👍